src/pages/modals/userModal/userModalHelpers.js

```diff
@@ -0,0 +1,88 @@
+import { getAge, isObjectFieldsEmpty } from '../../../scripts/helpers';
+import { emailRegular, phoneRegular } from '../../../scripts/regulars';
+import { MODAL_VALUES, USER_MODAL_TITLES } from '../../../constants/libraries';
+
+const formFields = {
+  name: '',
+  surname: '',
+  email: '',
+  direction: '',
+  sex: '',
+  role: '',
+  password: '',
+  confirmPassword: '',
+  birthDate: '',
+  address: '',
+  phone: '',
+  skype: '',
+  startDate: '',
+  education: '',
+  averageScore: '',
+  mathScore: '',
+};
+
+export const initStartUserModalState = {
+  ...formFields,
+  modalTitle: '',
+  readOnly: false,
+  formErrors: {
+    ...formFields,
+  },
+};
+
+export const getUserModalErrors = (state) => {
+  const { email, password, confirmPassword, birthDate, phone } = state;
+  const formErrors = { ...formFields };
+  const keys = Object.keys(formErrors);
+  keys.forEach((key) => {
+    if (!state[key].trim()) {
+      formErrors[key] = `${MODAL_VALUES[key]} is required`;
+    }
+  });
+
+  if (!emailRegular.test(email)) {
+    formErrors.email = 'Email is incorrect';
+  }
+  if (password.length < 8 || password.length > 24) {
+    formErrors.password = 'Password should contains 8-24 symbols';
+  } else if (password !== confirmPassword) {
+    formErrors.confirmPassword = 'Passwords are not equal';
+  }
+  if (!phoneRegular.test(phone)) {
+    formErrors.phone = 'Mobile phone is incorrect';
+  }
+  if (birthDate) {
+    const age = getAge(birthDate);
+    if (age < 18) {
+      formErrors.birthDate = 'User age is lower than 18';
+    }
+  }
+
+  if (!isObjectFieldsEmpty(formErrors)) {
+    return formErrors;
+  }
+
+  return null;
+};
+
+export const getReadUserModalState = (user) => {
+  return { ...initStartUserModalState, ...user, modalTitle: USER_MODAL_TITLES.read, readOnly: true };
+};
+
+export const getEditUserModalState = (user) => {
+  return { ...initStartUserModalState, ...user, modalTitle: USER_MODAL_TITLES.edit, readOnly: false };
+};
+
+export const getCreateUserModalState = () => {
+  return { ...initStartUserModalState, modalTitle: USER_MODAL_TITLES.create, readOnly: false };
+};
```

Comment:
export const getUserModalState = (title, readOnly = false) => {
  return { ...initStartUserModalState, title, readOnly };
};
`getUserModalState(user, USER_MODAL_TITLES.edit)`

Reply:
```suggestion
const getModalTitle = (user, readOnly) => {
  if (readOnly && user) {
    return USER_TITLES.read;
  }
  if (user) {
    return USER_TITLES.edit;
  }

  return USER_TITLES.read;
};

export const getUserModalState = (user = null, readOnly = false) => {
  const title = getModalTitle(user, readOnly);

  return { ...userModalState, ...user, title, readOnly };
};
```
And remove 3 mrthods from UserModal.jsx and looks better, thanks
```suggestion
componentDidMount() {
    const { user, readOnly } = this.props;
    const data = getUserModalState(user, readOnly);
    this.setState(data);
  }
```